code/modules/food_and_drink/cakes.dm

```diff
@@ -133,6 +133,8 @@
 				frostingoverlay.alpha = 255
 				src.UpdateOverlays(frostingoverlay,tag)
 				tube.reagents.trans_to(src,25)
+			if ((user.mind.assigned_role == "Chef") && (prob(25)))
+				JOB_XP(user, "Chef", 1)
```

Comment:
Applies for every change, but I think the `JOB_XP` macro already checks if their job matches before giving XP.